libpod/container_internal.go

```diff
@@ -260,12 +282,31 @@ func (c *Container) refresh() error {
 	if err != nil {
 		return errors.Wrapf(err, "error retrieving temporary directory for container %s", c.ID())
 	}
-	c.state.RunDir = dir
 
 	if err := c.runtime.state.SaveContainer(c); err != nil {
 		return errors.Wrapf(err, "error refreshing state for container %s", c.ID())
 	}
 
+	if len(c.config.IDMappings.UIDMap) != 0 || len(c.config.IDMappings.GIDMap) != 0 {
+		c.state.UserNSRoot, err = ioutil.TempDir("", fmt.Sprintf("libpod-%s", c.ID()))
+		if err != nil {
+			return errors.Wrapf(err, "error creating userNS tmpdir for container %s", c.ID())
+		}
+		rootUID, rootGID := c.RootUID(), c.RootGID()
+		if err := os.Chown(c.state.UserNSRoot, rootUID, rootGID); err != nil {
+			return err
+		}
+		if err = idtools.MkdirAllAs(c.state.DestinationRunDir, 0700, rootUID, rootGID); err != nil {
+			return errors.Wrapf(err, "cannot create userNS rundir for container %s", c.ID())
+		}
+	}
+
+	c.state.RunDir = dir
```

Comment:
Should probably move this block up above the SaveContainer().

Also, that `SaveContainer` should just be changed to `c.save()`, not sure how we missed that one.